### PR TITLE
[eclipse/xtext#1472] Change qualifier of target repository zip

### PR DESCRIPTION
**Pull request by kthoms**
For stable/release builds take the version as qualifier. Nightly can
stay as is.
- fixes bug#522525
- added method getBuildQualifier
- fix typo in method name generatePromoteProperties
- upgrade gradle wrapper to 5.4.1
- upgrade xtext-gradle-plugin

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
@kthoms there are conflicts + did you test that with a nightly?

**Comment by kthoms** [comment]
Have to look why this conflict exists. Mom...

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -1,11 +1,11 @@
 buildscript {
 	repositories.jcenter()
 	dependencies {
-		classpath 'org.xtext:xtext-gradle-plugin:1.0.15'
+		classpath 'org.xtext:xtext-gradle-plugin:2.0.4'
 	}
 }
 
-ext.xtextVersion = '2.10.0'
+ext.xtextVersion = '2.17.0'
```
Comment (cdietrich) [inline buildSrc/build.gradle]: 2.18.0?

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -16,6 +16,7 @@ repositories {
 }
 
 apply plugin: 'java'
+apply plugin: 'java-gradle-plugin'
```
Comment (cdietrich) [inline buildSrc/build.gradle]: why is this needed?

File: buildSrc/build.gradle (hunk 1 above)
Comment (kthoms) [reply buildSrc/build.gradle]: can do. does not really matter, but when we are already on it...

File: buildSrc/build.gradle (hunk 2 above)
Comment (kthoms) [reply buildSrc/build.gradle]: IIRC I had a problem when importing the project with buildship. Tried again and have no problem anymore without that. So seems obsolete and will be removed.

**Review by cdietrich — CHANGES_REQUESTED** [review]
nightly deploy got destroyed again

File: buildSrc/build.gradle (hunk 2 above)
Comment (cdietrich) [reply buildSrc/build.gradle]: still there

Comment (kthoms) [reply buildSrc/build.gradle]: removed

**Comment by cdietrich** [comment]
with the repo merge this is obsolete
